lib/service_manager.dart

```diff
@@ -139,114 +129,83 @@ class IsolateManager {
-    if (_flutterIsolate != null) {
-      _flutterIsolateController.add(_flutterIsolate);
-    }
-    _selectedIsolate = _selectBestFirstIsolate(isolates);
+    await _initSelectedIsolate(isolates);
     if (_selectedIsolate != null) {
       _isolateCreatedController.add(_selectedIsolate);
       _selectedIsolateController.add(_selectedIsolate);
+      // On initial connection to running app, service extensions are added from
+      // here.
+      await _serviceExtensionManager
+          ._addRegisteredExtensionRPCs(_selectedIsolate);
     }
   }
 
   void _handleIsolateEvent(Event event) async {
     if (event.kind == 'IsolateStart') {
       _isolates.add(event.isolate);
-      _flutterIsolates.add(event.isolate);
       _isolateCreatedController.add(event.isolate);
       if (_selectedIsolate == null) {
-        _selectedIsolate = event.isolate;
-        _selectedIsolateController.add(event.isolate);
+        _setSelectedIsolate(event.isolate);
       }
     } else if (event.kind == 'ServiceExtensionAdded') {
       // On hot restart, service extensions are added from here.
       await _serviceExtensionManager
           ._maybeAddServiceExtension(event.extensionRPC);
 
-      // Check to see if there is a new flutter isolate.
-      if (_flutterIsolate == null) {
-        if (_isFlutterExtension(event.extensionRPC)) {
-          _setFlutterIsolate(event.isolate);
-        }
+      // Check to see if there is a new isolate.
+      if (_selectedIsolate == null && _isFlutterExtension(event.extensionRPC)) {
+        _setSelectedIsolate(event.isolate);
       }
     } else if (event.kind == 'IsolateExit') {
       _isolates.remove(event.isolate);
-      _flutterIsolates.remove(event.isolate);
       _isolateExitedController.add(event.isolate);
       if (_selectedIsolate == event.isolate) {
         _selectedIsolate = _isolates.isEmpty ? null : _isolates.first;
         _selectedIsolateController.add(_selectedIsolate);
-      }
-      if (_flutterIsolate == event.isolate) {
-        _setFlutterIsolate(
-            _flutterIsolates.isEmpty ? null : _flutterIsolates.first);
         _serviceExtensionManager.resetAvailableExtensions();
       }
     }
   }
 
-  IsolateRef _selectBestFirstIsolate(List<IsolateRef> isolates) {
-    final IsolateRef ref = isolates.firstWhere((IsolateRef ref) {
-      // 'foo.dart:main()'
-      return ref.name.contains(':main(');
-    }, orElse: () => null);
-
-    if (ref != null) {
-      return ref;
-    }
-
-    return isolates.isEmpty ? null : isolates.first;
-  }
-
   bool _isFlutterExtension(String extensionName) {
     return extensionName.startsWith('ext.flutter.');
   }
 
-  Future<void> _initFlutterIsolate(List<IsolateRef> isolates) async {
+  Future<void> _initSelectedIsolate(List<IsolateRef> isolates) async {
     for (IsolateRef ref in isolates) {
-      // Populate flutter isolate info.
-      if (_flutterIsolate == null) {
+      if (_selectedIsolate == null) {
```

Comment:
We may want a notion of a list of flutter isolates, and a selected flutter isolate, and a separate notion of a list of all isolates, and a selected isolate. Right now we don't have the ability to have a selected isolate which is not a flutter isolate (I had been using a command-line app for testing).

Reply:
Would this be a sufficient solution? It would guarantee we always have a selected isolate even if none are flutter isolates.

After for loop:
`_selectedIsollate ??= isolates.isEmpty ? null : isolates.first;`

If that is not sufficient, it sounds like we probably need to revert the changes in this CL.

Reply:
Oh nevermind, I see you added this change to https://github.com/flutter/devtools/pull/80.